src/array/cuda/csr_mm.cu

```diff
@@ -118,7 +118,10 @@ std::pair<CSRMatrix, NDArray> CusparseSpgemm(
   CUSPARSE_CALL(cusparseDestroySpMat(matA));
   CUSPARSE_CALL(cusparseDestroySpMat(matB));
   CUSPARSE_CALL(cusparseDestroySpMat(matC));
-  return {CSRMatrix(A.num_rows, B.num_cols, dC_csrOffsets, dC_columns), dC_weights};
+  return {
+      CSRMatrix(A.num_rows, B.num_cols, dC_csrOffsets, dC_columns,
+                NullArray(dC_csrOffsets->dtype, dC_csrOffsets->ctx)),
```

Comment:
What's the benefit of specifying the dtype and ctx of a null array?

Reply:
The data array's dtype and ctx must be the same as that of indptr and indices even if the data array is a null one.